service/src/main/java/bio/terra/workspace/service/resource/referenced/ReferencedResourceService.java

```diff
@@ -68,7 +78,17 @@ public ReferencedResource createReferenceResourceForClone(
       ReferencedResource resourceToClone,
       ReferencedResource sourceReferencedResource,
       AuthenticatedUserRequest userRequest) {
+
+    validateDestinationWorkspacePolicies(
+        sourceReferencedResource.getWorkspaceId(),
+        resourceToClone.getWorkspaceId(),
+        sourceReferencedResource.getResourceType().getCloudPlatform());
     resourceDao.createReferencedResource(resourceToClone);
+    var result =
```

Comment:
I think this has to happen in a flight. If the `mergePao` fails, we need to delete the created referenced resource. We need it to behave transactionally.

There is no guarantee that the dry run test will be valid for the actual merge.